indent/lean.vim

```diff
@@ -0,0 +1,6 @@
+if exists("b:did_indent")
+  finish
+endif
+let b:did_indent = 1
+
+setlocal indentkeys+==end,==begin
```

Comment:
Can we make this file version-specific as well? begin/end no longer exists in Lean 4.

Reply:
Ok, so this is Lean 3-specific because it's lean and not lean4.  Then that's right.